build.sbt

```diff
@@ -43,6 +44,10 @@ lazy val commonSettings = Seq(
   scalacOptions in (Compile, doc) := (scalacOptions in (Compile, doc)).value.filter(_ != "-Xfatal-warnings")
 ) ++ warnUnusedImport
 
+lazy val machinistDependencies = Seq(
+  libraryDependencies += "org.typelevel" %%% "machinist" % "0.4.1"
```

Comment:
Right, the imp integration was just added in the last month or so. We could revert the imp integration to avoid the dependency and then tackle call-site inlining via macros ala imp and machinist once we start building typeclassic. I'm fine either way.